homework/index.js

```diff
@@ -31,16 +33,60 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
-    });
-  }
+    fetch(url)
+      .then(data => {
+        data.json()
+      })
+
+      .then(json => {
+        console.log(json)
+        const root = document.getElementById('root');
+        const select = createAndAppend('select', root, { class: 'select' });
+        createAndAppend('option', select, { text: 'Choose your favorite repo' });
+        json.forEach(repo => {
+          const name = repo.name;
+          createAndAppend('option', select, { text: name });
+
+          const repoInfo = createAndAppend('div', root);
+          const contribs = createAndAppend('div', root);
+          select.addEventListener('change', evt => {
+            const selectedRepo = evt.target.value;
+            const repo = json.filter(r => r.name == selectedRepo)[0];
+            console.log(repo);
+            repoInfo.innerHTML = '';
+            contribs.innerHTML = '';
+
 
+
+            const addInfo = (label, value) => {
+              const container = createAndAppend('div', repoInfo);
+              createAndAppend('span', container, { text: label });
+              createAndAppend('span', container, { text: value });
+            };
+            addInfo('Name: ', repo.name);
+            addInfo('Desciption: ', repo.description);
+            addInfo('Number of forks: ', repo.forks);
+            addInfo('Updated: ', repo.updated_at)
+            const contribsUrl = repo.contributors_url;
+            fetchJSON(contribsUrl)
+              .then(data.forEach(contributor => {
+
+                createAndAppend('div', contribs, { text: contributor.login, class: 'contributor' })
+                createAndAppend('img', contribs, { src: contributor.avatar_url, height: 100, widtth: 100, id: 'img' })
+              }))
+          })
+
+
+            .catch(err => {
+              createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            });
+        })
```

Comment:
There is some issue here with closing of functions `}` and/or `)` seems misplaced. When you fix the other issues the browser should give you an indication which are wrong.